meilisearch-types/src/error.rs

```diff
@@ -415,13 +425,28 @@ impl ErrorCode for milli::Error {
     }
 }
 
+impl ErrorCode for file_store::Error {
+    fn error_code(&self) -> Code {
+        match self {
+            Self::IoError(e) => e.error_code(),
+            Self::PersistError(e) => e.error_code(),
+        }
+    }
+}
+
+impl ErrorCode for tempfile::PersistError {
```

Comment:
I'm not very clear on what this implementation brings. Is this related to IoErrors?

Reply:
Yep, the `PersistError` can contain an `IoError` that must be handled properly